Add CITATION.cff

manics opened the pull request:
## :white_check_mark: Checklist

- [x] This pull request has a meaningful title.
- [x] If your changes are not yet ready to merge, you have marked this pull request as a **draft** pull request.

## :ballot_box_with_check: Maintainers' checklist

- [x] This pull request has had the appropriate labels assigned
- [x] This pull request has been added to the SATRE backlog project board
- [ ] This pull request has been assigned to one or more maintainers

### :arrow_heading_up: Summary

Adds a CITATION.cff file that tells people how to cite SATRE.

Combines authors from
    - https://zenodo.org/records/10055345
    - https://zenodo.org/records/17353472
    - https://github.com/sa-tre/satre-specification/issues/34#issuecomment-3750497144

### :closed_umbrella: Related issues

Closes https://github.com/sa-tre/satre-specification/issues/34

### :raising_hand: Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/sa-tre/satre-specification#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/sa-tre/satre-specification#contributors) in the README file: <!-- replace this text with the GitHub IDs of any new contributors -->

manics commented on CITATION.cff:
```diff
@@ -0,0 +1,21 @@
+message: Please cite SATRE using the metadata from this file
+type: software
+authors:
+  - {}
```
What should we put for the authors? The [current Zenodo record](https://zenodo.org/record/8233509) has `Ed Chalstrey; Jim Madge; James Robinson; Simon Li; Arron Lacey; harisood; Matt Craddock; Tim Machin`

gordonmilligan-uod commented:
> This hasn't been done. I think we should do it to encourage people to cite SATRE in a standard way, so maybe target 1.3 (based on [#365 (comment)](https://github.com/sa-tre/satre-specification/issues/365#issuecomment-2959064459)) ?

That makes sense to me. Shall move it.

manics committed:
Add CITATION.cff

Combines authors from
- https://zenodo.org/records/10055345
- https://zenodo.org/records/17353472
- https://github.com/sa-tre/satre-specification/issues/34#issuecomment-3750497144

manics committed:
Ensure CITATION.cff validates with https://github.com/citation-file-format/cffconvert

gordonmilligan-uod reviewed (approved):
LGTM - can't see anyobvious problems

drchriscole reviewed (approved):
Looks good. 

For version 2 might consider reviewing authorship.
